extensions/amp-ad-network-adsense-impl/0.1/test/test-amp-ad-network-adsense-impl.js

```diff
@@ -31,28 +48,177 @@ describe('amp-ad-network-adsense-impl', () => {
 
   beforeEach(() => {
     sandbox = sinon.sandbox.create();
-    adsenseImplElem = document.createElement('amp-ad');
-    adsenseImplElem.setAttribute('type', 'adsense');
-    adsenseImplElem.setAttribute('data-ad-client', 'adsense');
     sandbox.stub(AmpAdNetworkAdsenseImpl.prototype, 'getSigningServiceNames',
         () => {
           return ['google'];
         });
+    adsenseImplElem = getAdsenseImplElement({
+      'data-ad-client': 'adsense',
+      'width': '320',
+      'height': '50',
+      'data-experiment-id': '8675309',
+    });
     adsenseImpl = new AmpAdNetworkAdsenseImpl(adsenseImplElem);
   });
 
   afterEach(() => {
     sandbox.restore();
   });
 
+  describe('#getAdUrl', () => {
+
+    beforeEach(() => {
+      resetSharedState();
+    });
+
+    const invariantParams = {
+      'client': 'adsense',
+      'format': '320x50',
+      'w': '320',
+      'h': '50',
+      'output': 'html',
+      'is_amp': '3',
+      'eid': '8675309',
+    };
+    const variableParams = [
+      'slotname', 'adk', 'adf', 'ea', 'flash', 'url', 'wg', 'dt', 'bpp', 'bdt',
+      'fdt', 'idt', 'shb', 'cbv', 'saldr', 'amp_v', 'correlator', 'frm',
+      'ga_vid', 'ga_hid', 'iag', 'icsg', 'nhd', 'dssz', 'mdo', 'mso', 'u_tz',
+      'u_his', 'u_java', 'u_h', 'u_w', 'u_ah', 'u_aw', 'u_cd', 'u_nplug',
+      'u_nmime', 'dff', 'adx', 'ady', 'biw', 'isw', 'ish', 'ifk', 'oid', 'loc',
+      'rx', 'eae', 'pc', 'brdim', 'vis', 'rsz', 'abl', 'ppjl', 'pfx', 'fu',
+      'bc', 'ifi', 'dtd',
+    ];
+    // Skipping this test until all AdSense parameters are standardized, and
+    // their implementation in A4A and 3p reach parity.
+    it.skip('with single slot', () => {
```

Comment:
I fear we will never be able to include this test given a change to underlying adsense JS code would cause a breakage in the AMP runtime test coverage.  More likely we will need to make this part of an e2e test.  However, I see no negative in leaving it.